### ISSUE TITLE
Estimator from Keras Model fails to learn when original keras Model learns fine

### ISSUE THREAD

**Issue by erbian** [tensorflow/tensorflow#27924]
I have created a keras `Model` that learns a simple embedding from a deterministic function.  When I convert to an Estimator using ` tf.keras.estimator.model_to_estimator` learning fails to converge and the learned embeddings are incorrect.
  
I would expect that an Estimator constructed from a keras Model would have similar performance to the original keras Model.  However the keras model loss decays to zero quickly:

![image](https://user-images.githubusercontent.com/10658783/56299648-760e7a80-6102-11e9-92d0-600480fe8659.png)

But the Estimator does not learn at all:
![Screenshot from 2019-04-17 11-30-24](https://user-images.githubusercontent.com/10658783/56300844-a820dc00-6104-11e9-8fe8-fd4d7d70cbfa.png)

**Code to reproduce the issue**
```python
import numpy as np
import pandas as pd
import tensorflow as tf

# Generate Some Data
np.random.seed(1234)
n_obs = 200
n_sensors = 50
n_sources  = 5

Z = np.random.randn(n_obs, n_sources)
b = np.random.randn(n_sources, n_sensors)
y = Z.dot(b)

data = []
for t in range(n_obs):
    betas = pd.DataFrame(b).rename(lambda x: 'f_{}'.format(x)).T
    data.append(
        pd.DataFrame({'target': y[t,:], 't': t}).join(betas)
    )

data = pd.concat(data)

def build_keras_model():
"""
A model to learn `Z` in f(b, t) = Z[t].dot(b)
"""
    betas = tf.keras.layers.Input(shape=(n_sources,), name='betas')
    t = tf.keras.layers.Input(shape=(1,), name='t')

    sources = tf.keras.layers.Embedding(
        input_dim=n_obs,
        output_dim=n_sources,
        name='sources')(t)

    fitted = tf.keras.layers.Dot(axes=-1)([sources, betas])
    net = tf.keras.Model([betas, t], fitted)
    net.compile(optimizer='adadelta', loss='mse')
    return net

# Train Keras Model 
net = build_keras_model()
loss = []

hist = net.fit(
    x=[data.filter(like='f_'), data.t],
    y=data.target,
    batch_size=100,
    epochs=200)
loss += hist.history['loss']

#  get embedding weights and average squared different to ground truth
Z_hat_keras = net.get_layer('sources').get_weights()
print ((Z_hat_keras - Z) ** 2).mean()
# Out: 8.794945836110652e-05

# convert to a Estimator object
estimator = tf.keras.estimator.model_to_estimator(net)

# construct input functions for Estimator
def input_fn(data, batch_size, num_epochs, shuffle):
    ds = tf.data.Dataset.from_tensor_slices((
        {'betas': data.filter(like='f_'), 't': data.t}, data.target
    ))

    if shuffle:
        ds = ds.shuffle(len(data))

    return ds.repeat(num_epochs).batch(batch_size)

train_fn = lambda: input_fn(data, 100, 200, True)
eval_fn = lambda: input_fn(data, 500,1,False)

# run training on estimator
estimator.train(train_fn)
estimator.evaluate(eval_fn)
# Out: {'global_step': 20001, 'loss': 5.0131807}

#  get embedding weights and average squared different to ground truth
Z_hat_tensorflow = estimator.get_variable_value('sources/embeddings')
print ((Z_hat_tensorflow - Z) ** 2).mean()
# Out 0.928678746751657
```

**Other info / logs**
I have also posted on StackOverflow with a bit more detail:
https://stackoverflow.com/questions/55712219/are-keras-embedding-layers-getting-added-regularization-when-converted-to-tensor

**Comment by erbian** [comment]
Here is the output of the environment script:
[tf_env.txt](https://github.com/tensorflow/tensorflow/files/3090394/tf_env.txt)

**Comment by erbian** [comment]
The issue looks to be with the tensorflow optimizers.  The tensorflow version will not learn on identical keras models

![image](https://user-images.githubusercontent.com/10658783/56391791-b0a90d80-61fd-11e9-8a27-5b5f04213c23.png)

code for test:
```python
#  using build_keras_model defined in previous comment
net_w_keras_opt = build_keras_model()
net_w_keras_opt.compile(optimizer='adadelta', loss='mse')
net_w_tf_opt = build_keras_model()
net_w_tf_opt.compile(optimizer=tf.train.AdadeltaOptimizer(), loss='mse')

# save loss for plotting
keras_loss = []
tf_loss = []

# train keras
keras_hist = net_w_keras_opt.fit(
    x=[stacked.filter(like='source_'), stacked.sample_time],
    y=stacked.target,
    batch_size=50,
    epochs=200)
keras_loss += keras_hist.history['loss']

# train tf version
tf_hist = net_w_tf_opt.fit(
    x=[stacked.filter(like='source_'), stacked.sample_time],
    y=stacked.target,
    batch_size=50,
    epochs=200)
tf_loss += tf_hist.history['loss']

# plotting
%matplotlib inline
pd.Series(keras_loss).plot(label='Keras', legend=True)
pd.Series(tf_loss).plot(label='Tensorflow', legend=True, figsize=(10,5), grid=True)
```

**Comment by tanzhenyu** [comment]
which tf version is this? Can you try 1.14?

**Comment by ymodak** [comment]
The previous version was TF 1.13.1
With TF 1.14 we get similar performance.
![image](https://user-images.githubusercontent.com/42785357/60372074-35f42f80-99b0-11e9-9714-1f525ccb7a97.png)

**Comment by tanzhenyu** [comment]
I see you have put two different experiments:
1) experiment between keras model and estimator converted from it.
2) experiment between keras model using keras optimizer and tf optimizer.

For 1), can you try to reproduce it with 1.14?
For 2), this is expected, keras adadelta learning rate is 1.0, tf adadelta learning rate is 0.001. So latter learns much slower. You might want to adjust learning rate and test for consistency.

**Comment by lc0** [comment]
@tanzhenyu Would you mind to point me out where you can see learning rate for Adadelta is 0.001? 
I can see 1.0 for 1.14 at https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/optimizers.py#L381

please correct me if I am wrong

**Comment by lc0** [comment]
I think I found proper defaults from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/optimizer_v2/adadelta.py but in this case, it's pretty confusing and I am not sure what these old definitions for if they are not event exported

**Comment by tanzhenyu** [comment]
Not sure if I understand what do you mean...Does keras adadelta with learning rate of 1.0 help with your experiment?

**Comment by tomerk** [comment]
This appears to be a very old (almost 2 years, pre-dating tf 2.0) issue. So going ahead and closing. In general we strongly recommend avoiding estimators and using Keras directly.
